Release PR 1.0.0

jfy133 opened the pull request:
<!--
# nf-core/createtaxdb pull request

Many thanks for contributing to nf-core/createtaxdb!

Please fill in the appropriate checklist below (delete whatever is not relevant).
These are the most common things requested on pull requests (PRs).

Remember that PRs should be made against the dev branch, unless you're preparing a pipeline release.

Learn more about contributing: [CONTRIBUTING.md](https://github.com/nf-core/createtaxdb/tree/master/.github/CONTRIBUTING.md)
-->

## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/createtaxdb/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, also make a PR on the nf-core/createtaxdb _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core pipelines lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Check for unexpected warnings in debug mode (`nextflow run . -profile debug,test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

maxulysse commented on docs/images/nf-core-createtaxdb_logo_dark.png:
Ohh so you have 2 versions of the pipeline logo. Either with or without the dog.
Can't we just keep the one?

jfy133 replied:
Yeah agree

maxulysse commented on docs/usage/faq.md:
```diff
@@ -0,0 +1,272 @@
+We provide a list of required or recommended files, and which pipeline parameters to give them to here:
+
+- bracken
+  - taxonomy name dump file (`--namesdmp`)
+  - taxonomy nodes dump file (`--nodesdmp`)
```
this is an option for all tools. It is slightly confusing to repeat this in the docs this way.

maxulysse replied:
yeah, but only one option for multiple tools.
that's slightly confusing I believe

maxulysse commented on docs/usage/faq.md:
```diff
@@ -0,0 +1,272 @@
+
+### nucl2taxid
+
+An accession2taxid file is a file that maps database sequence accession IDs (e.g. NCBI GenBank) to their corresponding taxon IDs.
```
you mention `accession2taxid` but title of section is `nucl2taxid`

jfy133 replied:
It's serves the same purpose, but I've made a suggestion to make it clearer `accession2taxid-like file` (see below))

fbdtemme commented on workflows/createtaxdb.nf:
```diff
@@ -1,134 +1,310 @@
+    file_taxonomy_namesdmp // file: taxonomy names file
+    file_taxonomy_nodesdmp // file: taxonomy nodes file
+    file_accession2taxid   // file: accession2taxid file
+    file_nucl2taxid        // fle: nucl2taxid file
```
```suggestion
    file_nucl2taxid        // file: nucl2taxid file
```

jfy133 commented on assets/createtaxdb-metromap-diagram-dark.svg:
This is just an annoyinng SVG rendering issue because SVG is _apparently_ a 'format' so of course everyone will intepret it in the same way :roll_eyes: 

_Now without venting sarcasm: it's just a GitHub rendering thing, it opens fine in inkscape so the text objects are there - and regardless it's the PNGs which are the most important bit :) _

fbdtemme replied:
Aha, good to know.

jfy133 commented on assets/createtaxdb-metromap-diagram-light.svg:
See comment above (resolved), it's just a GitHub rendering issue

jfy133 commented on docs/usage/faq.md:
```diff
@@ -0,0 +1,272 @@
+  - taxonomy name dump file (`--namesdmp`)
+  - taxonomy nodes dump file (`--nodesdmp`)
+  - genome sizes file (`--genomesizes`)\*
+- kaiju (no additional files required)
```
```suggestion
- kaiju (no additional files required)
- kmcp (no additional files required))
```

maxulysse replied:
still need to update this

jfy133 commented on docs/usage/faq.md:
```diff
@@ -0,0 +1,272 @@
+
+### prot2taxid
+
+An accession2taxid file is a file that maps database sequence accession IDs (e.g. NCBI GenBank) to their corresponding taxon IDs.
```
```suggestion
An accession2taxid-like file that maps database sequence accession IDs (e.g. NCBI GenBank) to their corresponding taxon IDs.
```

jfy133 commented:
@maxulysse @fbdtemme thank you both for the reviews!

I think I've addressed all the comments. If you're happy please give a :heavy_check_mark: 

Then when I fix the test data issue I will merge in :)

jfy133 commented:
@nf-core-bot fix linting

maxulysse commented on tests/tags.yml:
no need for this file

maxulysse commented on workflows/createtaxdb.nf:
```diff
@@ -1,134 +1,310 @@
-    )
+    if (params.generate_tar_archive) {
+        TAR(ch_all_databases, '.gz')
+        TAR.out.archive
```
What is happening here?

maxulysse reviewed (changes requested):
minor comments

jfy133 commented on nf-test.config:
```diff
@@ -0,0 +1,24 @@
+
+    // load the necessary plugins
+    plugins {
+        load "nft-utils@0.0.3"
```
I want to keep as I suspect I will use it in the future

jfy133 commented:
@maxulysse PR resolving youre comments: https://github.com/nf-core/createtaxdb/pull/101

LilyAnderssonLee reviewed (commented):
I haven’t had time to go through everything. Since there are already two reviewers, I’ll skip the full review.
